### PR TITLE
candidate for #380 

### PR DESCRIPTION
**Pull request by cziaarm**
tl;dr Does what is required, but could perhaps be more efficient about it

# Story

IRUS sends work if when work id "investigated" `SharedBehaviorsController` u! :p  
a file_set id when file_set is "investigated" `FileSetsController` 
and file_set id when a file is "requested"  `DownloadsController`

but according to #380 IRUS considers these a single thing and is confounded by different IDs

# Expected Behavior Before Changes

Send IDs as above

# Expected Behavior After Changes

Send only the work id for works, file_sets and downloads

# Notes

To achieve the above I have used `AcitveFedora::Base.find(params[:id]).parent.id` This works, but hits fcrepo TWICE.

Happy to be availed of a nice way to do this :)

### CODE REVIEW

File: app/controllers/hyrax/downloads_controller.rb [hunk 1]
```diff
@@ -29,7 +29,12 @@ def show
     # OVERRIDE Hyrax 2.9.6 for IRUS Analytics
     def item_identifier_for_irus_analytics
       # return the OAI identifier
-      "#{CatalogController.blacklight_config.oai[:provider][:record_prefix].call(self)}:#{params[:id]}"
+      # We must send the parent work id to IRUS otherwise it gets confused
+      # This works but hits fcrepo twice :/
+      parent_work_id = ActiveFedora::Base.find(params[:id]).parent.id
```
Comment (jeremyf) [inline app/controllers/hyrax/downloads_controller.rb:34]: Using `ActiveFedora::Base.find` will pull from Fedora.  We may want to consider `ActiveFedora::Base.where(id: params[:id]).first`.

Further, the parent might be `nil` which would raise an exception.  We've found that the parent may not be set (see https://github.com/scientist-softserv/iiif_print/blob/d8246e1d95f340ae769dbd05be5a500930c9162a/app/services/iiif_print/pluggable_derivative_service.rb#L115-L119)

File: app/controllers/hyrax/file_sets_controller.rb [hunk 2]
```diff
@@ -80,7 +80,13 @@ def citation; end
     def item_identifier_for_irus_analytics
       # return the OAI identifier
       # http://www.openarchives.org/OAI/2.0/guidelines-oai-identifier.htm
-      "#{CatalogController.blacklight_config.oai[:provider][:record_prefix].call(self)}:#{params[:id]}"
+
+      # We must send the parent work id to IRUS otherwise it gets confused
+      # This works but hits fcrepo twice :/
+      parent_work_id = ActiveFedora::Base.find(params[:id]).parent.id
```
Comment (jeremyf) [inline app/controllers/hyrax/file_sets_controller.rb:86]: See above; sounds like we may want to consider a helper module to consolidate this logic.

File: app/controllers/hyrax/downloads_controller.rb (hunk 1 above)
Comment (cziaarm) [reply app/controllers/hyrax/downloads_controller.rb:34]: Aha... `where` rather than `find` I knew there must be a better way. Thanks @jeremyf !

**Comment by cziaarm** [comment]
@jeremyf Ready for re-review (if you have a mo)

File: app/controllers/hyrax/downloads_controller.rb [hunk 3]
```diff
@@ -29,7 +31,9 @@ def show
     # OVERRIDE Hyrax 2.9.6 for IRUS Analytics
     def item_identifier_for_irus_analytics
       # return the OAI identifier
-      "#{CatalogController.blacklight_config.oai[:provider][:record_prefix].call(self)}:#{params[:id]}"
+      # We must send the parent work id to IRUS otherwise it gets confused
+      parent_work_id = helpers.work_id_from_file_set_id(params[:id])
```
Comment (jeremyf) [inline app/controllers/hyrax/downloads_controller.rb:35]: What if you passed the controller and the id to the helper?  Then you could encapsulate the duplicate configuration logic.

File: app/helpers/hyrax/irus_helper.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module IrusHelper
+    def work_id_from_file_set_id(file_set_id)
```
Comment (jeremyf) [inline app/helpers/hyrax/irus_helper.rb:5]: What if the `where` clause returns an empty array?

File: app/helpers/hyrax/irus_helper.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      parent_for(ActiveFedora::Base.where(id: file_set_id).first).id
+    end
+
+    def parent_for(file_set)
+      file_set.parent || file_set.member_of.find(&:work?)
```
Comment (jeremyf) [inline app/helpers/hyrax/irus_helper.rb:10]: What if this method returns `nil`?

**Review by jeremyf — APPROVED** [review]

**Review by jeremyf — APPROVED** [review]
